client/blocks/plan-storage/index.jsx

```diff
@@ -60,9 +61,14 @@ export function PlanStorage( { children, className, siteId } ) {
 		) {
 			mediaStorage.max_storage_bytes = 1024 * 1024 * 1024;
 		}
+
 		if ( sitePlanSlug === PLAN_WPCOM_PRO ) {
 			mediaStorage.max_storage_bytes = 50 * 1024 * 1024 * 1024;
 		}
+
+		if ( sitePlanSlug === PLAN_WPCOM_STARTER ) {
+			mediaStorage.max_storage_bytes = 6 * 1024 * 1024 * 1024;
+		}
```

Comment:
I'm currently working on this file in #63394 as part of moving plan checks to feature checks (p4TIVU-a66-p2). 

Since `sitePlanSlug` is based on the current plan of the site and this is not about what features a plan has in general, do you think it would make sense to use a feature check here?

Or, if we don't want to show the highest available storage size for the site but specifically the storage site that comes with the current plan, could `planHasFeature` be used here to decouple the logic from checking specific plans?

Reply:
@chriskmnds @DavidRothstein Is there any advice or thoughts you have for me on this?

Reply:
Apologies @obenland . I was planning to get back to you, but got caught up in a hectic release/revert scenario with this PR.

I agree if we can elevate this to feature checks would be cleaner. Also not a fun of hardcoding feature values that happens here and elsewhere (the legacy storage limit of 3GB is added in the logic inline here, also in the plan storage  feature for the grid).

> if we don't want to show the highest available storage size for the site but specifically the storage site that comes with the current plan

This is not too clear to me (whether this is what we want to do or not). The code renders the available space seen in the media library, so it might be conceptualy, but maybe not techically right now ❔  a site-specific characteristic

I will make a comment in the p2 post. I wanted to ask why Calypso has so much knowledge of features and feature details offhand, when all of that could (should?) be coming from the API - for all i know, i was able to render the complete Starter plan features list (albeit incomplete plan spec, with missing prices) without having a Starter plan returned from the REST endpoint. That seemed wrong to me, so anything in direction of consolidating looks like a win.

Reply:
> I wanted to ask why Calypso has so much knowledge of features and feature details offhand, when all of that could (should?) be coming from the API - for all i know, i was able to render the complete Starter plan features list (albeit incomplete plan spec, with missing prices) without having a Starter plan returned from the REST endpoint. That seemed wrong to me, so anything in direction of consolidating looks like a win.

I don't _know_ but that doesn't keep me from assuming :) I could see it having started out as a way to power the plan comparison page with feature information, which then developed in the source of truth for plan <=> feature association for Calypso.

We'd _love_ to have that be powered by the features endpoint eventually, so everything that happens in `wpcom-features` automatically is available in Calypso as well.

Reply:
> We'd love to have that be powered by the features endpoint eventually, so everything that happens in `wpcom-features` automatically is available in Calypso as well.

Absolutely. Thanks for reply. Here's a bit more surface on the issue from a similar angle (where these feature descriptions/meta are used - and potentially in more places that I missed): https://github.com/Automattic/wp-calypso/issues/63764